### PR TITLE
fix: [FX-2739] Fix circularity in Button files - part 1

### PR DESCRIPTION
**Pull request by denieler**
[FX-2739]

### Description

Fix circularity issues in `Button` component files

### How to test

- `yarn circularity`
-  see 47 errors instead of 53

### Screenshots

Screenshot 2022-05-16 at 11 32 28

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2739]: https://toptal-core.atlassian.net/browse/FX-2739?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso/src/ButtonCompound/index.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import Radio from '../ButtonRadio'
+
+export const ButtonCompound = Object.assign(Button, {
+  Group,
+  Circular,
+  Action,
+  Split,
+  Checkbox,
+  Radio
+})
```
Comment (dmaklygin) [inline packages/picasso/src/ButtonCompound/index.tsx:16]: 👍

File: .madgerc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "fileExtensions": ["js", "ts", "jsx", "tsx"],
+  "tsConfig": "./tsconfig.json"
+}
```
Comment (dmaklygin) [inline .madgerc:4]: it also take into account `picasso/dist-package/Button/index.js`, so I would add `excludeRegExp` option, to exclude such folders. wdyt?

**Review by dmaklygin — APPROVED** [review]

File: .prettierignore [hunk 3]
```diff
@@ -1,28 +1,3 @@
...
-packages/picasso/src/NativeSelectOptions/index.tsx
-packages/picasso/src/NativeSelectPlaceholder/index.tsx
-packages/picasso/src/Select/index.ts
-packages/picasso/src/TimePicker/index.ts
-packages/picasso/src/utils/Transitions/index.ts
-packages/picasso/src/YearSelect/index.ts
-packages/picasso/src/Utils/index.ts
-packages/picasso/src/PageSidebar/index.ts
-packages/picasso/src/SidebarItem/index.ts
-packages/topkit-analytics-charts/src/AnalyticsChart/index.ts
```
Comment (separatio) [inline .prettierignore:24]: I had removed all of this once, but it was re-added back for some reason. :smile: Thanks for removing them again.

**Review by separatio — APPROVED** [review]
Changes look good. PR is missing changeset.

**Review by ozgurkececioglu — APPROVED** [review]

**Comment by denieler** [comment]
> Changes look good. PR is missing changeset.

@separatio this PR should not change how users use Button component, so I haven't added changeset because of that

File: packages/picasso/src/ButtonGroup/test.tsx [hunk 4]
```diff
@@ -1,16 +1,16 @@
 import React from 'react'
 import { render } from '@toptal/picasso/test-utils'
 
-import Button from '../Button'
+import ButtonGroup from '../ButtonGroup'
```
Comment (separatio) [inline packages/picasso/src/ButtonGroup/test.tsx:4]: Seems like ButtonGroup usage is changed. Don't we need a changeset for that? :thinking: 

Staff Portal is using `Button.Group`, for example.

Comment (denieler) [reply packages/picasso/src/ButtonGroup/test.tsx:4]: @separatio no, they would use it as they used it before. None of stories have been changed

**Review by dmaklygin — APPROVED** [review]

**Review by TomasSlama — APPROVED** [review]

**Pull request merged**